### PR TITLE
[DomCrawler] Allow using non-absolute base URIs

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12318
| License       | MIT
| Doc PR        | -

@xabbuh @stof I implemented in this PR your comments from https://github.com/symfony/symfony/issues/12318

### CODE REVIEW

**Comment by javiereguiluz** [comment]
In the latest commit 1f797ea21603c7c5f9fbeb11f7825cecffa46eb1 I made the constructor argument optional, because this is probably what the issue reporter wanted in the first place. If we agree with this ... we should rebase into master as a new feature.

**Review by stof — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@stof do you agree that we should rebase this on `master` and treat it as a new feature? Thanks!

**Comment by stof** [comment]
We could indeed consider that a new feature rather than a bugfix.

**Comment by xabbuh** [comment]
IMO there is one downside with this new approach: Previously, you could safe call `getUri()` as the object was in a valid state. Now you could run into exceptions based on the internal state. That's probably something consuming code isn't prepared for.

**Comment by javiereguiluz** [comment]
@xabbuh you are right! I've refactored this to introduce the new feature without changing the current behavior much. Thanks!

File: src/Symfony/Component/DomCrawler/AbstractUriElement.php [hunk 1]
```diff
@@ -40,15 +40,17 @@
...
-        if (!\in_array(strtolower(substr($currentUri, 0, 4)), array('http', 'file'))) {
-            throw new \InvalidArgumentException(sprintf('Current URI must be an absolute URL ("%s").', $currentUri));
-        }
-
         $this->setNode($node);
         $this->method = $method ? strtoupper($method) : null;
         $this->currentUri = $currentUri;
+
+        $elementUriIsRelative = null === parse_url(trim($this->getRawUri()), PHP_URL_SCHEME);
+        $baseUriIsAbsolute = \in_array(strtolower(substr($this->currentUri, 0, 4)), array('http', 'file'));
```
Comment (xabbuh) [inline src/Symfony/Component/DomCrawler/AbstractUriElement.php]: Should we use `strpos()` here instead?

Comment (javiereguiluz) [reply src/Symfony/Component/DomCrawler/AbstractUriElement.php]: It's the same code as in the older branches, so we know it works and I'd prefer to not change it 😅

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @javiereguiluz.

**Pull request merged**